Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/TasksPage.java

```diff
@@ -79,12 +80,16 @@ public TasksPage goTo() throws Exception {
 
     public List<String> getTaskDetails() {
         int index = triggerRowToggle(taskTable, "Progress");
-        WebElement detailsTable1 = Browser.getDriver()
-                .findElementById(TASK_TABLE + ":" + index + ":taskDetailTableFirst");
-        WebElement detailsTable2 = Browser.getDriver()
-                .findElementById(TASK_TABLE + ":" + index + ":taskDetailTableSecond");
-        List<String> taskDetails = getTableDataByColumn(detailsTable1, 1);
-        taskDetails.addAll(getTableDataByColumn(detailsTable2, 1));
+        String firstElementId = TASK_TABLE + ":" + index + ":taskDetailTableFirst";
+        String secondElementId = TASK_TABLE + ":" + index + ":taskDetailTableSecond";
+        WebElement firstDetails = Browser.getDriver().findElementById(firstElementId);
+        WebElement secondDetails = Browser.getDriver().findElementById(secondElementId);
+        await("Wait for first task details to become visible").atMost(3, TimeUnit.SECONDS)
+                .untilAsserted(() -> assertTrue(Browser.getDriver().findElement(By.id(firstElementId)).isDisplayed()));
+        List<String> taskDetails = getTableDataByColumn(firstDetails, 1);
+        await("Wait for second task details to become visible").atMost(3, TimeUnit.SECONDS)
+                .untilAsserted(() -> assertTrue(Browser.getDriver().findElement(By.id(secondElementId)).isDisplayed()));
```

Comment:
I think  if  this two variables will be declared after `assertTrue`  is better.
```suggestion
        await("Wait for first task details to become visible").atMost(3, TimeUnit.SECONDS)
                .untilAsserted(() -> assertTrue(Browser.getDriver().findElement(By.id(firstElementId)).isDisplayed()));
                WebElement firstDetails = Browser.getDriver().findElementById(firstElementId);
        List<String> taskDetails = getTableDataByColumn(firstDetails, 1);
        await("Wait for second task details to become visible").atMost(3, TimeUnit.SECONDS)
                .untilAsserted(() -> assertTrue(Browser.getDriver().findElement(By.id(secondElementId)).isDisplayed()));
                WebElement secondDetails = Browser.getDriver().findElementById(secondElementId);
```